nengo_interfaces/airsim.py

```diff
@@ -25,76 +35,257 @@ class AirSim(nengo.Process):
+            if "training_mode" in camera_params:
+                warnings.warn(
+                    "'training_mode' key is deprecated, use 'use_physics' key"
+                    + " instead to toggle physics engine"
+                )
+                if "use_physics" not in camera_params:
+                    camera_params["use_physics"] = camera_params["training_mode"]
+
+            self.use_physics = self.camera_params["use_physics"]
+
+            if "capture_settings" not in camera_params:
+                camera_params["capture_settings"] = None
+
+            # create the capture save folder if it doesn't exist
+            _tmp = camera_params["save_name"]
+            if _tmp is not None:
+                if not os.path.exists(_tmp):
+                    os.makedirs(_tmp)
+            _tmp = None
+
+        # update our settings.json that airsim reads in
+        home = expanduser("~")
+        with open(  # pylint: disable=W1514
+            f"{home}/Documents/AirSim/settings.json", "r+"
+        ) as fp:
+            data = json.load(fp)
+            prev_data = copy.deepcopy(data)
+
+            # physics mode, set to multirotor
+
+            CUR_DIR = os.path.realpath(os.path.join(os.path.dirname(__file__), "."))
+            if self.use_physics:
+                data["SimMode"] = "Multirotor"
+                data["EngineSound"] = True
+                print("phys mode")
+
+                if "Vehicles" not in data.keys():
+                    # load vehicle params from backup file
+                    with open(  # pylint: disable=W1514
+                        f"{CUR_DIR}/fly_settings.json", "r+"
+                    ) as fp_fly:
+                        # save vehicle params to backup file
+                        vehicle_data = json.load(fp_fly)["Vehicles"]
+                        data["Vehicles"] = vehicle_data
+
+            # computer vision, turn off physics to speed things up
+            else:
+                data["SimMode"] = "ComputerVision"
+                data["EngineSound"] = False
+
+                if "Vehicles" in data.keys():
+                    with open(  # pylint: disable=W1514
+                        f"{CUR_DIR}/fly_settings.json", "r+"
+                    ) as fp_fly:
+                        # save vehicle params to backup file
+                        vehicle_data = json.load(fp_fly)
+                        vehicle_data["Vehicles"] = data["Vehicles"]
+                        fp_fly.seek(0)
+                        json.dump(vehicle_data, fp_fly, indent=4)
+                        fp_fly.truncate()
+
+                    # remove from dict as this will break ComputerVision mode
+                    data.pop("Vehicles")
+
+            # can turn display off for speed boost
+            if not show_display:
+                data["ViewMode"] = "NoDisplay"
+            else:
+                # fly with me for drone, spring chase for cars
+                data["ViewMode"] = ""
```

Comment:
If the ViewMode was originally set to something else, this will overwrite it. (e.g. sometimes you might want "SpringArmChase" for a drone). This should probably be left as what was in `prev_data` unless it was "NoDisplay", which gets a little tricky because the old "ViewMode" needs to be backed up somewhere.